### PR TITLE
Fix: Adding comments on ingredients parenting to make it clear what is being done [MTT-8377]

### PR DESCRIPTION
**Pull request by bastienunity**
### Description
This PR removes the SetPhysics method from ServerIngredientPhysics and is adding comments to the changes made to the Rigidbody variables to make the intent easier to understand.

### Issue Number(s)
[MTT-8377](https://jira.unity3d.com/browse/MTT-8377) Adding a comment on why Rigidbody.interpolation need to be changed on kinematic objects when parenting them.

### Contribution checklist
 - [ ] Tests have been added for the project and/or any internal package
 - [ ] Release notes have been added to the [project changelog](../CHANGELOG.md) file
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] JIRA ticket ID is in the PR title or at least one commit message
 - [x] Include the ticket ID number within the body message of the PR to create a hyperlink

### CODE REVIEW

**Comment by unity-cla-assistant** [comment]
[CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=165) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=165) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=165) it.</sub>

File: Basic/ClientDriven/Assets/Scripts/ServerIngredientPhysics.cs [hunk 1]
```diff
@@ -14,13 +13,24 @@ public class ServerIngredientPhysics : NetworkBehaviour
...
-        SetPhysics(parentNetworkObject == null);
-    }
-
-    void SetPhysics(bool isEnabled)
-    {
-        m_Rigidbody.isKinematic = !isEnabled;
-        m_Rigidbody.interpolation = isEnabled ? RigidbodyInterpolation.Interpolate : RigidbodyInterpolation.None;
-        m_NetworkTransform.InLocalSpace = !isEnabled;
+        // The physic needs to be disabled when the ingredient is parented so that it's following the parent instead of moving freely.
+        if (parentNetworkObject != null)
```
Comment (RikuTheFuffs) [inline Basic/ClientDriven/Assets/Scripts/ServerIngredientPhysics.cs:17]: nitpick: in general, having a positive-phrased condition makes reading easier vs having a negative-phrased (I.E: "X is true" is easier to read than "X is not false"), so I'd swap the if and else branches.

File: Basic/ClientDriven/Assets/Scripts/ServerIngredientPhysics.cs [hunk 2]
```diff
@@ -14,13 +13,24 @@ public class ServerIngredientPhysics : NetworkBehaviour
...
+        {
+            // Setting isKinematic to true will prevent the object from being affected by the physic in the game,
```
Comment (RikuTheFuffs) [inline Basic/ClientDriven/Assets/Scripts/ServerIngredientPhysics.cs:19]: As per the company's coding conventions, multi-line comments should be done with `/**/` (applies to other comments as well)

**Review by RikuTheFuffs — CHANGES_REQUESTED** [review]
this PR introduces a lot of code duplication, which is not good as every future change would have to be duplicated.

Since the old approach was less-readeable mainly because the logic used to invert the `isEnabled` flag all the time, wouldn't it make more sense to just invert the flag and call it `disablePhysics` to make it more descriptive?

**Comment by bastienunity** [comment]
@RikuTheFuffs 
>Since the old approach was less-readeable mainly because the logic used to invert the isEnabled flag all the time, wouldn't it make more sense to just invert the flag and call it disablePhysics to make it more descriptive?
Good suggestion, I've put the code back into a method and named it SetKinematic to follow Unity's naming on the rigidoby and avoid the extra negative boolean we used to have.

File: Basic/ClientDriven/Assets/Scripts/ServerIngredientPhysics.cs [hunk 3]
```diff
@@ -14,13 +13,18 @@ public class ServerIngredientPhysics : NetworkBehaviour
 
     public override void OnNetworkObjectParentChanged(NetworkObject parentNetworkObject)
     {
-        SetPhysics(parentNetworkObject == null);
+        // The ingredient's physic needs to be disabled when the ingredient is parented so that it's following the parent instead of moving freely.
+        SetKinematic(parentNetworkObject != null);
     }
 
-    void SetPhysics(bool isEnabled)
+    private void SetKinematic(bool isEnabled)
```
Comment (RikuTheFuffs) [inline Basic/ClientDriven/Assets/Scripts/ServerIngredientPhysics.cs:20]: nitpick (no need to fix it): `private` is implicit in C#, we don't have to write it. In fact, by not writing it, codes is faster  to read (as there's less stuff) and it becomes easier to distinguish between public and private method with a glympse

Comment (bastienunity) [reply Basic/ClientDriven/Assets/Scripts/ServerIngredientPhysics.cs:20]: Oh, I think my rider setting just add them automatically, I'll fix that for the next PR, thanks!

**Pull request merged**